### PR TITLE
Feat: add address and user profile page.

### PR DESCRIPTION
**Pull request by Bhakti199**
here, I added userprofile page and address management.

### CODE REVIEW

File: src/ApiCalls/addOrdersCall.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      {
+        order,
+      },
+      {
+        headers: {
+          authorization: localStorage.getItem("userLoginToken"),
+        },
+      }
+    );
+    console.log(data, status);
```
Comment (rkapoor10) [inline src/ApiCalls/addOrdersCall.js:16]: remove console.log if not needed

File: src/ApiCalls/postNewUser.js [hunk 2]
```diff
@@ -8,6 +8,7 @@ export const postNewUser = async (firstName, lastName, email, password) => {
       email,
       password,
     });
+    console.log(data, status);
```
Comment (rkapoor10) [inline src/ApiCalls/postNewUser.js:11]: here also

File: src/ApiCalls/removeAddressCall.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import axios from "axios";
+
+export const removeAddressCall = async (addressId) => {
+  try {
+    const { data, status } = await axios.delete(
+      `/api/user/addresses/${addressId}`,
+      {
+        headers: {
+          authorization: localStorage.getItem("userLoginToken"),
+        },
```
Comment (rkapoor10) [inline src/ApiCalls/removeAddressCall.js:10]: you may use axios config if you want not to pass the token again and again

File: src/Pages/UserProfilePage/UserProfilePage.css [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  min-height: 100vh;
+}
+.user-profile {
+  /* border: 2px solid var(--black-color); */
+  /* border-radius: 100%; */
+  /* padding: 12px 0; */
+  display: flex;
```
Comment (rkapoor10) [inline src/Pages/UserProfilePage/UserProfilePage.css:11]: remove comments

**Review by rkapoor10 — APPROVED** [review]
- very well structured code
- implementation is good
- go through some clean up comments
- apart from this looks good to merge

File: src/Pages/UserProfilePage/ProfileSubComponents/Address/Address.jsx [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+import { useState } from "react";
+export const Address = () => {
+  const [showAddressForm, setShowAddressForm] = useState(false);
+  const { addAddress, removeAddress, setAddressForOrder, user } = useUser();
+  const { addresses, firstName, lastName } = user;
+
+  const addressFormHandler = (event) => {
+    event.preventDefault();
+    const [mobileNumber, area, city, state, pincode] = event.target.elements;
+    console.log(mobileNumber.value.length, pincode.value.length);
```
Comment (tushartiwari7) [inline src/Pages/UserProfilePage/ProfileSubComponents/Address/Address.jsx:16]: remove console

File: src/Pages/UserProfilePage/ProfileSubComponents/Address/Address.jsx [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+            />
+          </div>
+          <button type="submit" className="login-form-btn">
+            Add Address
+          </button>
+        </form>
+      )}
+      <div className="address-row-container">
+        {addresses &&
+          addresses.length > 0 &&
```
Comment (tushartiwari7) [inline src/Pages/UserProfilePage/ProfileSubComponents/Address/Address.jsx:88]: can use `address?.length` instead. 
Same thing, lesser no. of lines.

**Review by tushartiwari7 — APPROVED** [review]
Code is well structured and clean. Good to merge

**Pull request merged**